experimental/generation/generator/packages/library/src/mergeAssets.ts

```diff
@@ -294,148 +292,189 @@ async function mergeRootFile(schemaName: string, oldPath: string, oldFileList: s
-            const entityLUContent = listEntitySectionName + os.EOL + resultStatements.join(os.EOL)
-            updatedLUResource = oldSectionOp.updateSection(oldListEntitySection.Id, entityLUContent)
+/**
+ * @description: Get the set of deleted utterance patterns.
+ * @param filename Name of the lu file.
+ * @param newFileList List of old file paths.
+ * @param delUtteranceSet Set of deleted utterance patterns.
+ */
+async function getDeletedUtteranceSet(filename: string, oldFileList: string[], delUtteranceSet: Set<string>): Promise<void> {
+    const filePath = oldFileList.filter(file => file.match(filename))[0]
+    const text = await fs.readFile(filePath, 'utf8')
+    const lines = text.split(os.EOL)
+    for (let line of lines) {
+        if (line.startsWith('>') && line.match(valuePattern)) {
+            line = line.replace('>', '')
+            const newLine = await generatePatternUtterance(line)
+            delUtteranceSet.add(newLine)
+        }
+    }
+}
 
-            // update intent content
-            if (oldIntentSections.length === 0) {
-                continue
+/**
+ * @description: Update generated lu file if the old lu file has been modified.
+ * @param filename Name of the lu file.
+ * @param newFileList List of new file paths.
+ * @param newPath Path to the folder of the new asset.
+ * @param mergedPath Path to the folder of the merged asset.
+ * @param delUtteranceSet Set of deleted utterance patterns.
+ * @param feedback Callback function for progress and errors.
+ */
+async function updateGeneratedLuFile(filename: string, newFileList: string[], newPath: string, mergedPath: string, delUtteranceSet: Set<string>, feedback: Feedback): Promise<void> {
+    const filePath = newFileList.filter(file => file.match(filename))[0]
+    const text = await fs.readFile(filePath, 'utf8')
+    const lines = text.split(os.EOL)
+    const resultLines: string[] = []
+    for (const line of lines) {
+        if (line.match(valuePattern)) {
+            const newLine = await generatePatternUtterance(line)
+            if (delUtteranceSet.has(newLine)) {
+                resultLines.push(`>${line}`)
+            } else {
+                resultLines.push(line)
             }
+        } else {
+            resultLines.push(line)
+        }
+    }
+    let val = resultLines.join(os.EOL)
+    await writeToFile(newPath, mergedPath, filename, newFileList, val, feedback)
+}
 
-            const oldIntentSection = oldIntentSections[0]
-            const removedEnumValue = new Set<string>()
-            for (const enumSyn of deletedEnumValue) {
-                if (!keepEnumValue.has(enumSyn)) {
-                    removedEnumValue.add(enumSyn)
-                }
-            }
+/**
+ * @description: Generate the common pattern of the utterances.
+ * @param line Line of the lu file.
+ */
+async function generatePatternUtterance(line: string): Promise<string> {
+    line = line.replace('-', '').trim()
```

Comment:
>line.replace('-', '').trim() [](start = 11, length = 28)

Too aggressive.  What would happen if I had the string foo-blah in my text?  I think I led you astray because I've been thinking C# rather than node.  

line.replace(/{@?[^=]+=|}|^>+\s*-?\s*|^-\s*|^\s*|\s*$/gm, '') will generate the utterance and you should do this any line that starts with '>'. It will also handle the - > and spaces at start/'end.